process/block/metablock.go

```diff
@@ -1854,106 +1779,60 @@ func (mp *metaProcessor) MarshalizedDataToBroadcast(
 	return mrsData, mrsTxs, nil
 }
 
-func (mp *metaProcessor) getSortedHeadersPerShard(round uint64) (map[uint32][]*hashAndHdr, int, error) {
-	shardBlocksPool := mp.dataPool.ShardHeaders()
-	if shardBlocksPool == nil {
-		return nil, 0, process.ErrNilShardBlockPool
-	}
-
-	mp.mutNotarizedHdrs.RLock()
-	if mp.notarizedHdrs == nil {
-		mp.mutNotarizedHdrs.RUnlock()
-		return nil, 0, process.ErrNotarizedHdrsSliceIsNil
-	}
-
-	hashAndBlockMap := make(map[uint32][]*hashAndHdr)
-	for _, key := range shardBlocksPool.Keys() {
-		val, _ := shardBlocksPool.Peek(key)
-		if val == nil {
-			continue
-		}
+func (mp *metaProcessor) getLongestShardsChainFromLastNotarized(round uint64) ([]data.HeaderHandler, [][]byte, map[uint32][]data.HeaderHandler, error) {
```

Comment:
I think the whole function could go inside blockTracker.

Reply:
ComputeLongestChain from block tracker returns a list of sorted headers/hashes for a given shard and started form a given header. You may not want all the longest chain headers for all the shards. Here we have 4 types returned. I would not complicate the block tracker and make another method ComputeLongestChainForAllShards only for this case. More than this you should give to this method your last notarized header for which you want to start the computation of the longest chain and you do not want to use the block tracker last notarized as it would be always higher than the one that you are processing right now.

Reply:
BlockTracker would implement this function as well, with the same input and output. The thing is, all the data and processing inside the method is using only blockTracker data -> it can be done there.

Reply:
Done